### PR TITLE
chore(common): CHP-7101 prepare repo for monorepo

### PR DESCRIPTION
**Pull request by chanceaclark**
## What/why?

- Convert over to yarn in preparation for yarn workspaces + lerna.
- Use node v14 and npm v6.
- Update to the latest dependencies (all majors were node version support).

_Note: this is being merged into a `v3` branch._

### CODE REVIEW

**Comment by chanceaclark** [comment]
CircleCI has an outage right now, but here's a screenshot in the meantime:
Screen Shot 2021-07-07 at 09 56 11

File: package.json [hunk 1]
```diff
@@ -19,13 +19,14 @@
     "test": "jest"
   },
   "engines": {
-    "node": ">=10"
+    "node": "^14.0.0",
+    "npm": "^6.0.0"
```
Comment (deini) [inline package.json:23]: This will be a warning for everyone using npm 7 🤔 do we want that?

**Review by deini — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (chanceaclark) [reply package.json:23]: I'm okay with that, since npm 7 is still a little buggy afaik

Comment (chanceaclark) [reply package.json:23]: Updated in #51

Comment (icatalina) [reply package.json:23]: I don't think this tool should be restricted to use with `npm@6` thought. What's the benefit in that?

Comment (chanceaclark) [reply package.json:23]: We updated it to `>=6` in #50. Do you think we should support lower versions of npm?

Comment (icatalina) [reply package.json:23]: No, that's fine, missed the other change 🙇‍♂️

**Pull request merged**